### PR TITLE
add non parametric layernorm, 'FastLayerNorm', as faster/safer option than RMSNorm

### PR DESCRIPTION
**Pull request by lessw2020**
This PR adds the option for users to run non-parametric LayerNorm, aka FastLayerNorm. 
Based on the OLMo paper, they did extensive norm testing and found that non-parametric layer norm (no scaling or gain) was both the fastest and safest norm option (vs regular LayerNorm or RMSNorm).
(see https://arxiv.org/html/2402.00838v1)

This is now controlled in llama-> modelargs as 'norm_type', and can toggle between fastlayernorm or rmsnorm (str). 

Testing:
ran with fastlayernorm and saw slightly better loss curve under debug model vs RMSNorm (quick test though) but at least confirms it trains fine.  Missing my other other PR for avg iter timing but can benchmark later on larger model, main goal here was it runs and has equal or better loss curve.

### CODE REVIEW

File: torchtrain/models/llama/model.py [hunk 1]
```diff
@@ -393,8 +419,18 @@ def __init__(self, layer_id: int, model_args: ModelArgs):
         )
         self.layer_id = layer_id
         self.num_layers = model_args.n_layers
-        self.attention_norm = RMSNorm(model_args.dim, eps=model_args.norm_eps)
-        self.ffn_norm = RMSNorm(model_args.dim, eps=model_args.norm_eps)
+        if model_args.norm_type == "rmsnorm":
+            self.attention_norm = RMSNorm(model_args.dim, eps=model_args.norm_eps)
+            self.ffn_norm = RMSNorm(model_args.dim, eps=model_args.norm_eps)
+        elif model_args.norm_type == "fastlayernorm":
```
Comment (tianyu-l) [inline torchtrain/models/llama/model.py:425]: shall we also add an if-statement to the [code](https://github.com/pytorch/torchtrain/blob/main/torchtrain/parallelisms/parallelize_llama.py#L142) which registers RMSNorm computation? Basically only `distribute_rmsnorm` if using "rmsnorm".

File: torchtrain/models/llama/model.py [hunk 2]
```diff
@@ -27,6 +27,29 @@ class ModelArgs:
     depth_init: bool = (
         True  # initialization uses each unique layer_id or total model layer count
     )
+    norm_type: str = "fastlayernorm"  # or "rmsnorm"
+
+
+class FastLayerNorm(torch.nn.Module):
```
Comment (wanchaol) [inline torchtrain/models/llama/model.py:33]: If we want to support different normalization layers, we should:
1. make those norm implementations into a separate file (`norm.py`), instead of living in model.py
2. expose a `build_normalization_layer` API in that module and use it in model.py
3. make sure SP covers this layer
4. probably make this layer named as `NonParametricLayerNorm` directly
